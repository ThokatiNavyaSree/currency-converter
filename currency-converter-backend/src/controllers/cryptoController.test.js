const axios = require('axios');
const { getTopCryptos, validateInput, convertCurrency } = require('./cryptoController');

jest.mock('axios');

describe('Currency Converter Backend Tests', () => {
  it('should fetch top cryptocurrencies successfully', async () => {
    const mockResponse = {
      data: [{ id: 'bitcoin', name: 'Bitcoin' }],
    };

    axios.get.mockResolvedValue(mockResponse);

    const req = {};
    const res = { json: jest.fn() };

    await getTopCryptos(req, res);

    expect(res.json).toHaveBeenCalledWith(mockResponse.data);
    expect(res.json).toMatchSnapshot();
  });


  it('should convert currency successfully', async () => {
    const mockConversionData = {
      data: {
        bitcoin: { usd: 50000 },
      },
    };

    axios.get.mockResolvedValue(mockConversionData);

    const req = { query: { sourceCrypto: 'bitcoin', amount: 1, targetCurrency: 'usd' } };
    const res = { json: jest.fn() };

    await convertCurrency(req, res);

    expect(res.json).toHaveBeenCalledWith({ convertedAmount: 50000 });
    expect(res.json).toMatchSnapshot();
  });
});
