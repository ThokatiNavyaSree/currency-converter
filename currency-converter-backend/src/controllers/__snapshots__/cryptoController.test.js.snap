// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency Converter Backend Tests should convert currency successfully 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "convertedAmount": 50000,
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Currency Converter Backend Tests should fetch top cryptocurrencies successfully 1`] = `
[MockFunction] {
  "calls": [
    [
      [
        {
          "id": "bitcoin",
          "name": "Bitcoin",
        },
      ],
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
